### ISSUE TITLE
Help download speed very low

### ISSUE THREAD

**Issue by cyberpunkbln** [neonmoe/minreq#86]
Hello,

when i use minreq as an downloader of binary-datas from an url, it is very slow.

standard call:

```
let response = minreq::get(Uri);
let response = response.send().ok().unwrap();
```

lazy and normal mode from the Request. Only send and the protocol the speed while retrieving the body in memory.

**Comment by neonmoe** [comment]
Some level of slowness is to be expected, as it isn't my intention to make an especially fast, but a small HTTP client. Is it slow compared to something else specifically? Is it still slow when tested with different servers?

**Comment by cyberpunkbln** [comment]
@neonmoe 
Hello, so i have made two rust-prgs, one with your Library and absolute minimal code and with antoher library http-req.

minreq = around 8Mbit/s and 32% cpu load
http-req = around 300Mbit/s and 6% cpu load

https://ftp.halifax.rwth-aachen.de/mxlinux/isos/ANTIX/Final/antiX-22/antiX-22_386-base.iso
http://ftp.halifax.rwth-aachen.de/mxlinux/isos/ANTIX/Final/antiX-22/antiX-22_386-base.iso

with the pure examples from doc.rs

**Comment by neonmoe** [comment]
That is more of a difference than I would've thought! There's definitely room for optimization in this crate, but that's honestly a bit embarassing, losing by a factor of 40. I'll leave this open for future investigation, but I don't currently have the time to work on this, so I'd suggest going with http-req in your case!

**Comment by cyberpunkbln** [comment]
@neonmoe its a pity, i like your minimal library, i will look in the future.

**Comment by neonmoe** [comment]
This should be fixed in the newest version, 2.7.0. Thanks for reporting the issue!
